skywalking-sniffer/skywalking-sdk-plugin/mongodb-plugin/src/main/java/com/a/eye/skywalking/plugin/mongodb/MongoDBReadBindingInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package com.a.eye.skywalking.plugin.mongodb;
+
+import com.a.eye.skywalking.api.context.ContextManager;
+import com.a.eye.skywalking.api.plugin.interceptor.EnhancedClassInstanceContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodInvokeContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.MethodInterceptResult;
+import com.mongodb.ServerAddress;
+import com.mongodb.binding.ReadBinding;
+
+/**
+ * {@link MongoDBReadBindingInterceptor} record the host and port information from {@link EnhancedClassInstanceContext#context},
+ *
+ * @author baiyang
+ */
+public class MongoDBReadBindingInterceptor implements InstanceMethodsAroundInterceptor {
```

Comment:
Process first into the execute method, and then into the  getReadBinding method or getWriteBinding method. the getReadBinding method or getWriteBinding method will return the ReadBinding or WriteBinding .I want to get the Host and Port info from them. Process sequence can be identified, execute-> getReadBinding or getWriteBinding -> other method. MongoDBMethodInterceptor intercept  execute method, So i create span in MongoDBMethodInterceptor, set the host/port in MongoDBReadBindingInterceptor or MongoDBWriteBindingInterceptor.  
At first, I want to intercept the com.mongodb.Mongo#Mongo(final Cluster cluster, final MongoClientOptions options, final List<MongoCredential> credentialsList) Constructor to get the host/port info, in  the onConstruct method i can get host/port info and set them into EnhancedClassInstanceContext. but i can not get the host/port in the MongoDBMethodInterceptor class from the EnhancedClassInstanceContext. So i must get host/port from getReadBinding method or getWriteBinding method.